### DISCUSSION TITLE
converting epoch to datetime?

### DISCUSSION THREAD

**Discussion by JeffMelton** [nushell/nushell#11933] [Q&A] [answered]
I found [this discussion](https://github.com/nushell/nushell/discussions/10469), but it's not quite what I'm looking for. I have a log timestamp formatted like `1708613219.56`, and I'd like to convert that to a human-readable date (something like `1708613219.56 | date humanize`). From what I can tell so far, nushell's datetime operations don't like floats, but I can't figure out how to convert that into a format nushell _does_ like. Any help is much obliged. :)

**Comment by fdncred** [comment, marked as answer]
We typically convert these numbers to nanoseconds and use `number | into datetime`.

Is this close?
```nushell
❯ 1708613219560000000 | into datetime
Thu, 22 Feb 2024 14:46:59 +0000 (3 hours ago)
```
Or with a different format
```nushell
❯ 1708613219560000000 | into datetime | format date "%+"
2024-02-22T14:46:59.560+00:00
```
Or humanized
```nushell
❯ 1708613219560000000 | into datetime | date humanize
3 hours ago
```
One more way using math
```nushell
❯ 1708613219.56 * 1_000_000_000 | into int | into datetime
Thu, 22 Feb 2024 14:46:59 +0000 (3 hours ago)
```

**Reply by JeffMelton** [reply]
Yep, that's what I'm looking for. Not sure why I couldn't wrap my head around the conversion I needed, but I appreciate the help. 🎉

**Reply by JeffMelton** [reply]
And for completeness in case someone else comes along:
```
1708613219.56 * (10 ** 9) | into int | into datetime | date humanize
```
